Source/Csla.Blazor.WebAssembly/ApplicationContextManager.cs

```diff
@@ -11,180 +11,177 @@
-    {
-      _currentPrincipal = null;
-      AuthenticationState = task;
-    }
+  private void AuthenticationStateProvider_AuthenticationStateChanged(Task<AuthenticationState> task)
+  {
+    AuthenticationState = task;
+  }
 
-    /// <summary>
-    /// Gets a value indicating whether this
-    /// context manager is valid for use in
-    /// the current environment.
-    /// </summary>
-    public bool IsValid => true;
-
-    /// <summary>
-    /// Gets a value indicating whether the current runtime
-    /// is stateful (e.g. WPF, Blazor, etc.)
-    /// </summary>
-    public bool IsStatefulContext => true;
-
-    /// <summary>
-    /// Gets the current principal.
-    /// </summary>
-    public IPrincipal GetUser()
-    {
-      if (_currentPrincipal == null)
-      {
-        if (AuthenticationState.IsCompleted && AuthenticationState.Result != null)
-          _currentPrincipal = AuthenticationState.Result.User;
-        else
-          _currentPrincipal = new ClaimsPrincipal();
-      }
-      return _currentPrincipal;
-    }
+  /// <summary>
+  /// Gets a value indicating whether this
+  /// context manager is valid for use in
+  /// the current environment.
+  /// </summary>
+  public bool IsValid => true;
 
-    /// <summary>
-    /// Sets the current principal ONLY IN APPLICATIONCONTEXT.
-    /// To set the value correctly, use your specific
-    /// ApplicationContextStateProvider implementation.
-    /// </summary>
-    /// <param name="principal">Principal object.</param>
-    public virtual void SetUser(IPrincipal principal)
-    {
-      throw new NotSupportedException(nameof(SetUser));
-    }
+  /// <summary>
+  /// Gets a value indicating whether the current runtime
+  /// is stateful (e.g. WPF, Blazor, etc.)
+  /// </summary>
+  public bool IsStatefulContext => true;
+
+  private static readonly IPrincipal _anonymous = new ClaimsPrincipal(new ClaimsIdentity());
+
+  /// <summary>
+  /// Gets the current principal.
+  /// </summary>
+  public IPrincipal GetUser()
+  {
+    IPrincipal result;
+    if (AuthenticationState.IsCompleted && AuthenticationState.Result != null)
+      result = AuthenticationState.Result.User;
+    else
+      result = _anonymous;
+    return result;
+  }
+
+  /// <summary>
+  /// Sets the current principal ONLY IN APPLICATIONCONTEXT.
+  /// To set the value correctly, use your specific
+  /// ApplicationContextStateProvider implementation.
+  /// </summary>
+  /// <param name="principal">Principal object.</param>
+  public virtual void SetUser(IPrincipal principal)
+  {
+    throw new NotSupportedException(nameof(SetUser));
+  }
 
-    /// <summary>
-    /// Gets the local context.
-    /// </summary>
-    public IContextDictionary GetLocalContext()
+  /// <summary>
+  /// Gets the local context.
+  /// </summary>
+  public IContextDictionary GetLocalContext()
+  {
+    IContextDictionary localContext;
+    var sessionManager = ApplicationContext.GetRequiredService<ISessionManager>();
```

Comment:
I know not part of this PR. But is there any reason why `ISessionManager` can't be a dependency provided through the constructor?

Reply:
Agreed, I'll make this change

Reply:
LOL! Or not, as that would result in a circular DI dependency